### PR TITLE
dynasty transition

### PR DESCRIPTION
**Pull request by djrtwo**
Addresses #89

### CODE REVIEW

**Comment by hwwhww** [comment]
Good job! ✨
Will review it (my) tonight!

File: beacon_chain/state/state_transition.py [hunk 1]
```diff
@@ -322,6 +324,64 @@ def fill_recent_block_hashes(active_state: ActiveState,
...
+
+
+def compute_dynasty_transition(crystallized_state: CrystallizedState,
+                               block: 'Block',
+                               config: Dict[str, Any]=DEFAULT_CONFIG) -> CrystallizedState:
+    crystallized_state = deepcopy(crystallized_state)
+    crystallized_state.current_dynasty += 1
+
+    # Not current in spec, but should be added soon
+    crystallized_state.dynasty_start = crystallized_state.last_state_recalc
```
Comment (hwwhww) [inline beacon_chain/state/state_transition.py]: The spec is updated to `Set last_dynasty_start = block.slot_number`
(should be `crystallized_state.dynasty_start`)

File: beacon_chain/state/state_transition.py [hunk 2]
```diff
@@ -322,6 +324,64 @@ def fill_recent_block_hashes(active_state: ActiveState,
...
+    # check that crosslinks were updated for all required shards
+    required_crosslinks_updated = True
+    for shard_id, crosslink in enumerate(crystallized_state.crosslink_records):
+        if shard_id in required_shards:
+            required_crosslinks_updated &= crosslink.slot > crystallized_state.dynasty_start
+
+
+    slots_since_last_dynasty_change = block.slot_number - crystallized_state.dynasty_start
+
+    return slots_since_last_dynasty_change >= config['min_dynasty_length'] and \
```
[the remark this answers is not in the document]
Comment (djrtwo) [reply beacon_chain/state/state_transition.py]: Yeah, we should immediately return when we get anything False in this function and should definitely leave the shard check for last.

Thhanks

File: beacon_chain/state/state_transition.py (hunk 1 above)
Comment (djrtwo) [reply beacon_chain/state/state_transition.py]: spec changed back after mini debate with V :)

**Pull request merged**